manager-bundle/src/EventListener/PreviewAuthenticationListener.php

```diff
@@ -64,6 +71,10 @@ public function __invoke(RequestEvent $event): void
             return;
         }
 
-        $event->setResponse(new RedirectResponse($this->router->generate('contao_backend_login')));
+        $event->setResponse(
+            new RedirectResponse($this->uriSigner->sign($this->router->generate('contao_backend_login', [
+                'redirect' => $request->getUri(),
```

Comment:
how come we can use `$request->getUri()` here?

Reply:
Same as above.